packages/compass-import-export/src/modules/import.ts

```diff
@@ -625,16 +597,15 @@ export const setDelimiter = (delimiter: CSVDelimiter) => {
       delimiter: delimiter,
     });
 
-    if (previewLoaded) {
-      debug('preview needs updated because delimiter changed', {
+    // TODO: This is certainly a race condition. What if it is busy loading?
```

Comment:
I'm thinking it is probably better to just always dispatch loadCSVPreviewDocs() again. Ideally we would abort the previous attempt, but that's probably overkill. The new attempt should most likely finish after the previous one.

Thoughts?

Reply:
It should be pretty fast right? I think a good comment is enough in that case. If there's a good chance `loadCSVPreviewDocs()` will be slow lets track the promise and abort/discard it then.

Reply:
OK I'm making it re-update regardless of previewLoaded and I updated the comment.